### PR TITLE
[zh] Switch language name 'zh' to 'zh-cn'

### PR DESCRIPTION
**Pull request by tengqm**
This is the first step to rename 'zh' to 'zh-cn'. There are several reasons why we rename the language name.

- The upstream docsy theme changed the language name, leading to many warnings during site build;
  The side-effect is that the i18n strings are no longer working.
- We believe renaming the language is the right thing to do, because this move can make room for other variants of Chinese language, such as 'zh-tw', 'zh-sg' etc.

There would be several follow-ups to this PR, such as fixing the intra-site links, adding redirects etc.

We will lock up changes to zh/zh-cn pages for the moment, until this one gets in.

This PR is based on commit cdad0a7342abccc1e13b9750c8c523a6946ff026.

Related: #33955

### CODE REVIEW

**Comment by howieyuen** [comment]
cool
/approve

**Review by sftim — CHANGES_REQUESTED** [review]
https://deploy-preview-34215--kubernetes-io-main-staging.netlify.app/zh/ doesn't redirect to https://deploy-preview-34215--kubernetes-io-main-staging.netlify.app/zh-cn/

Let's add that redirect, it's an important detail to preserve existing bookmarks any hyperlinks.

I'd use a 302 redirect because in the future we might want to change where this redirects to.

**Comment by tengqm** [comment]
@sftim I'm planning to add 'zh' redirection in follow-ups.

This PR only renames the directory. We have a lot of zh -> zh-cn conversions (especially those links prefixed by '/zh/') to be converted.

**Comment by sftim** [comment]
If you add a one-line redirect into `static/_redirects`, the existing links to `/zh/*` still work (albeit with an extra request and response to get redirected). I definitely think it's worth adding.

**Comment by tengqm** [comment]
@sftim Added as suggested.

File: static/_redirects [hunk 1]
```diff
@@ -18,7 +18,7 @@
 /ru/docs/     /ru/docs/home/ 301!
 /uk/docs/     /uk/docs/home/ 301!
 /vi/docs/     /vi/docs/home/ 301!
-/zh/docs/     /zh/docs/home/ 301!
+/zh/docs/     /zh-cn/docs/home/ 301!
```
Comment (sftim) [inline static/_redirects:21]: (nit)
The `/zh-cn/` redirect is perhaps more important.
```suggestion
/zh/docs/     /zh-cn/docs/home/ 301!
/zh-cn/docs/     /zh-cn/docs/home/ 301!
```

**Review by sftim — APPROVED** [review]
/hold cancel

File: static/_redirects (hunk 1 above)
Comment (tengqm) [reply static/_redirects:21]: done.

**Pull request merged**